jpyinterpreter/src/main/java/org/optaplanner/jpyinterpreter/implementors/JumpImplementor.java

```diff
@@ -50,13 +57,39 @@ public static void popAndJumpIfFalse(MethodVisitor methodVisitor, int jumpTarget
         methodVisitor.visitJumpInsn(Opcodes.IF_ACMPEQ, jumpLocation);
     }
 
+    /**
+     * Pops TOS. If TOS is not None, set the bytecode counter to {@code jumpTarget}.
+     */
+    public static void popAndJumpIfIsNotNone(FunctionMetadata functionMetadata, StackMetadata stackMetadata, int jumpTarget) {
+        MethodVisitor methodVisitor = functionMetadata.methodVisitor;
+        Map<Integer, Label> bytecodeCounterToLabelMap = functionMetadata.bytecodeCounterToLabelMap;
+
+        Label jumpLocation = bytecodeCounterToLabelMap.computeIfAbsent(jumpTarget, key -> new Label());
+        PythonConstantsImplementor.loadNone(methodVisitor);
+        methodVisitor.visitJumpInsn(Opcodes.IF_ACMPNE, jumpLocation);
+    }
+
+    /**
+     * Pops TOS. If TOS is None, set the bytecode counter to {@code jumpTarget}.
+     */
+    public static void popAndJumpIfIsNone(FunctionMetadata functionMetadata, StackMetadata stackMetadata, int jumpTarget) {
+        MethodVisitor methodVisitor = functionMetadata.methodVisitor;
+        Map<Integer, Label> bytecodeCounterToLabelMap = functionMetadata.bytecodeCounterToLabelMap;
+
+        Label jumpLocation = bytecodeCounterToLabelMap.computeIfAbsent(jumpTarget, key -> new Label());
+        PythonConstantsImplementor.loadNone(methodVisitor);
+        methodVisitor.visitJumpInsn(Opcodes.IF_ACMPEQ, jumpLocation);
+    }
+
     /**
      * TOS and TOS1 are an exception types. If TOS1 is not an instance of TOS, set the bytecode counter to the
      * {@code instruction} argument.
      * Pop TOS and TOS1 off the stack.
      */
```

Comment:
The code uses "TOS" a lot, but this description is somewhat hidden deep inside the class. Maybe a class-level Javadoc would work better here?